geomstats/geometry/product_manifold.py

```diff
@@ -179,7 +179,9 @@ def embed_to_product(self, points):
                 points_.append(response)
 
             return gs.concatenate(points_, axis=-1)
-        return gs.stack(points, axis=-2)
+        if len(self.shape) <= 2:
+            return gs.stack(points, axis=-2)
```

Comment:
the `if` is unnecessary, the return below is enough.

Reply:
Done.